packages/story-utils/src/getStories.ts

```diff
@@ -10,13 +11,21 @@ type StoryData = {
   storyFn: Story<unknown>;
 };
 
+const metadataSchema = zod
+  .object({
+    title: zod.string(),
+    parameters: zod.record(zod.unknown()).optional(),
+  })
+  .nonstrict();
+
 export default function getStories(globPath = '**/*.stories.*'): StoryData[] {
   const filePaths = glob.sync(globPath);
 
   return filePaths.flatMap((filePath) => {
     // eslint-disable-next-line @typescript-eslint/no-var-requires
     const storiesFileExports = require(path.resolve(filePath));
-    const { default: metadata, ...storiesMap } = storiesFileExports;
+    const { default: rawMetadata, ...storiesMap } = storiesFileExports;
+    const metadata = metadataSchema.parse(rawMetadata);
 
     return Object.keys(storiesMap).map((storyName: string) => {
       const story = storiesMap[storyName] as Story<unknown>;
```

Comment:
Ideally we'd validate these stories as well.

Not sure if it's possible to validate a function + some additional properties. I've asked the question, though: https://github.com/colinhacks/zod/issues/244